src/solver/optimisation/constraint_builder.h

```diff
@@ -0,0 +1,206 @@
+      case Variable::NumberBreakingDownDispatchableUnits:
+        return nativeOptimVar.NumeroDeVariableDuNombreDeGroupesQuiTombentEnPanneDuPalierThermique[index_];
+      case Variable::NTCDirect:
+        return nativeOptimVar.NumeroDeVariableDeLInterconnexion[index_];
+      case Variable::IntercoDirectCost:
+        return nativeOptimVar.NumeroDeVariableCoutOrigineVersExtremiteDeLInterconnexion[index_];
+      case Variable::IntercoIndirectCost:
+        return nativeOptimVar.NumeroDeVariableCoutExtremiteVersOrigineDeLInterconnexion[index_];
+      case Variable::ShortTermStorageInjection:
+        return nativeOptimVar.SIM_ShortTermStorage.InjectionVariable[index_];
+      case Variable::ShortTermStorageWithdrawal:
+        return nativeOptimVar.SIM_ShortTermStorage.WithdrawalVariable[index_];
+      case Variable::ShortTermStorageLevel:
+        return nativeOptimVar.SIM_ShortTermStorage.LevelVariable[index_];
+      case Variable::HydProd:
+        return nativeOptimVar.NumeroDeVariablesDeLaProdHyd[index_];
+      case Variable::HydProdDown:
+        return nativeOptimVar.NumeroDeVariablesVariationHydALaBaisse[index_];
+      case Variable::HydProdUp:
+        return nativeOptimVar.NumeroDeVariablesVariationHydALaHausse[index_];
+      case Variable::Pumping:
+        return nativeOptimVar.NumeroDeVariablesDePompage[index_];
+      case Variable::HydroLevel:
+        return nativeOptimVar.NumeroDeVariablesDeNiveau[index_];
+      case Variable::Overflow:
+        return nativeOptimVar.NumeroDeVariablesDeDebordement[index_];
+      case Variable::FinalStorage:
+        return problemeHebdo.NumeroDeVariableStockFinal[index_];
+      case Variable::LayerStorage:
+        return problemeHebdo.NumeroDeVariableDeTrancheDeStock[index_][0]; // TODO FIXMEEE
+      case Variable::PositiveUnsuppliedEnergy:
+        return nativeOptimVar.NumeroDeVariableDefaillancePositive[index_];
+      case Variable::NegativeUnsuppliedEnergy:
+        return nativeOptimVar.NumeroDeVariableDefaillanceNegative[index_];
+      default:
+        return -1;
+    }
+    }
+
+public:
+    ConstraintBuilder& include(Variable var, double coeff, int timeShift = 0, bool wrap = false)
+    {
+      std::vector<double>& Pi = problemeAResoudre.Pi;
+      std::vector<int>& Colonne = problemeAResoudre.Colonne;
+      int varIndex = getVariableIndex(var, timeShift, wrap);
+      if (varIndex >= 0)
+      {
+        Pi[nombreDeTermes_] = coeff;
+        Colonne[nombreDeTermes_] = varIndex;
+        nombreDeTermes_++;
+      }
+      return *this;
+    }
+    ConstraintBuilder& equal(double rhs)
+    {
+      operator_ = '=';
+      rhs_ = rhs;
+      return *this;
+    }
+    ConstraintBuilder& less(double rhs)
+    {
+      operator_ = '<';
+      rhs_ = rhs;
+      return *this;
+    }
+    ConstraintBuilder& greater(double rhs)
+    {
+      operator_ = '>';
+      rhs_ = rhs;
+      return *this;
+    }
+
+    void build();
```

Comment:
Maybe this could be called directly when defining operator and rhs, but not sure.